src/main/java/io/supertokens/webserver/api/oauth/OAuthClientsAPI.java

```diff
@@ -0,0 +1,168 @@
+import io.supertokens.pluginInterface.RECIPE_ID;
+import io.supertokens.pluginInterface.Storage;
+import io.supertokens.pluginInterface.exceptions.InvalidConfigException;
+import io.supertokens.pluginInterface.exceptions.StorageQueryException;
+import io.supertokens.pluginInterface.multitenancy.AppIdentifier;
+import io.supertokens.pluginInterface.multitenancy.exceptions.TenantOrAppNotFoundException;
+import io.supertokens.webserver.InputParser;
+import io.supertokens.webserver.WebserverAPI;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+import java.io.Serial;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+public class OAuthClientsAPI extends WebserverAPI {
+
+    @Serial
+    private static final long serialVersionUID = -4482427281337641246L;
+
+    private static final List<String> ALLOWED_INPUT_FIELDS = Arrays.asList(new String[]{"clientName","scope", "redirectUris", "allowedCorsOrigins", "authorizationCodeGrantAccessTokenLifespan", "authorizationCodeGrantIdTokenLifespan", "authorizationCodeGrantRefreshTokenLifespan",
+        "clientCredentialsGrantAccessTokenLifespan","implicitGrantAccessTokenLifespan","implicitGrantIdTokenLifespan","refreshTokenGrantAccessTokenLifespan","refreshTokenGrantIdTokenLifespan","refreshTokenGrantRefreshTokenLifespan","tokenEndpointAuthMethod","audience",
+        "grantTypes","responseTypes","clientUri","logoUri","policyUri","tosUri","metadata"});
+    private static final List<String> REQUIRED_INPUT_FIELDS = Arrays.asList(new String[]{"clientName", "scope"});
+
+    @Override
+    public String getPath() {
+        return "/recipe/oauth/clients";
+    }
+    public OAuthClientsAPI(Main main){
+        super(main, RECIPE_ID.OAUTH.toString());
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
+        JsonObject input = InputParser.parseJsonObjectOrThrowError(req);
+
+        boolean containsAllRequired = containsAllRequiredFields(input);
+        boolean containsMoreThanAllowed = containsMoreThanAllowed(input);
+        if(!containsAllRequired || containsMoreThanAllowed){
+            throw new ServletException(new WebserverAPI.BadRequestException("Invalid Json Input"));
+        }
+
+        try {
+            AppIdentifier appIdentifier = getAppIdentifier(req);
+            Storage storage = enforcePublicTenantAndGetPublicTenantStorage(req);
+
+            JsonObject response = OAuth.registerOAuthClient(super.main, appIdentifier, storage, input);
+            sendJsonResponse(200, response, resp);
+
+        } catch (OAuthClientRegisterException registerException) {
+
+            JsonObject errorResponse = createJsonFromException(registerException);
+            sendJsonResponse(400, errorResponse, resp);
```

Comment:
response needs to be correctly formatted based on how we have apis in the core like:
```
{status: "CLIENT_ID_ALREADY_EXISTS_ERROR"|
```

With a status of 200

Reply:
based on todays discussion, in this case of client already exists, the core should silentlry retry. that's done.